service/awsconfig/v10/resource/s3bucket/create.go

```diff
@@ -45,6 +45,24 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 			return microerror.Mask(err)
 		}
 
+		err = r.createAccessLogBucket(ctx, bucketInput)
+		if err != nil {
+			return microerror.Mask(err)
+		}
```

Comment:
Creating the extra bucket here is OK. But I think it would be better to extend the resource so it can create multiple S3 buckets.

This is what we do for the s3object resource.

https://github.com/giantswarm/aws-operator/blob/087f506cfc669ca05ae25e84c295e01c8870e6a0/service/awsconfig/v10/resource/s3object/spec.go#L16-L20 

The desired spec for the bucket can include the logging settings.

Reply:
Ok so it does mean creating a new resource folder, add create/delete/update an instance and pass it to the framework?

Reply:
A new resource isn't needed. It means extending the current resource so it can create multiple S3 buckets. This is how the s3object resource already works.

So instead of having.

```
type BucketState struct {
	Name string
}
```

You have.

```
map[string]BucketState{}
```

The BucketState can be extended to have the extra logging config.

Reply:
ahh ok get it now! ;)